lib/Main.js

```diff
@@ -422,8 +422,11 @@ Main.prototype.onMatrixEvent = function(ev) {
         // A membership event about myself
         var membership = ev.content.membership;
         if (membership === "invite") {
-            // Automatically accept all invitations
-            this.getBotIntent().join(ev.room_id);
+            // Automatically accept all invitations, but wait 30 seconds before doing it.
+            // this was throwing an error when the bot was invited to a room by a user on
+            // a different homeserver ({"errcode":"M_FORBIDDEN","error":"You are not invited to this room."})
+            Promise.delay(30 * 1000).then(() => this.getBotIntent().join(ev.room_id));
+
```

Comment:
Empty line.

And this feels like a bug in Synapse :s. I wonder if it can be reproduced with a normal client or if it is related to the appservice API.